File: backend/models/store.js
const db = require("../config/database");
const bcrypt = require("bcrypt");

class Store {
  static createStore(storeData, callback) {
    const {
      storeName,
      storeEmail,
      storePhone,
      storeAddress,
      storeLatitude,
      storeLongitude,
      storePassword,
    } = storeData;

    bcrypt.hash(storePassword, 10, (err, hash) => {
      if (err) {
        callback(err, null);
      } else {
        const insertStoreQuery =
          "INSERT INTO `stores`(store_name,store_address,store_latitude,store_longitude,store_phone,store_email,store_password) VALUES (?,?,?,?,?,?,?)";
        db.query(
          insertStoreQuery,
          [
            storeName,
            storeAddress,
            storeLatitude,
            storeLongitude,
            storePhone,
            storeEmail,
            hash,
          ],
          (err, result) => {
            if (err) {
              callback(err, null);
            } else {
              callback(null, result);
            }
          }
        );
      }
    });
  }

  static getStore(email, password, callback) {
    const getStoreQuery = "SELECT * FROM `stores` WHERE ?";
    db.query(getStoreQuery, [email], (err, result) => {
      if (err) {
        callback(err, null);
      } else if (results.length > 0) {
        const store = results[0];
        bcrypt.compare(password, store.store_password, (err, result) => {
          if (result) {
            callback(null, store);
          } else {
            callback(null, null);
          }
        });
      } else {
        callback(null, null);
      }
    });
  }
}

module.exports = Store;

File: lib/models/store.dart
class Store {
  int? storeid;
  String? storename;
  String? storeaddress;
  double? storelatitude;
  double? storelongitude;
  String? storephone;
  String? storeemail;
  String? storepassword;

  Store(
      {this.storeid,
      this.storename,
      this.storeaddress,
      this.storelatitude,
      this.storelongitude,
      this.storephone,
      this.storeemail,
      this.storepassword});

  Store.fromJson(Map<String, dynamic> json) {
    storeid = json['store_id'];
    storename = json['store_name'];
    storeaddress = json['store_address'];
    storelatitude = json['store_latitude'];
    storelongitude = json['store_longitude'];
    storephone = json['store_phone'];
    storeemail = json['store_email'];
    storepassword = json['store_password'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = Map<String, dynamic>();
    storeid = data['store_id'];
    storename = data['store_name'];
    storeaddress = data['store_address'];
    storelatitude = data['store_latitude'];
    storelongitude = data['store_longitude'];
    storephone = data['store_phone'];
    storeemail = data['store_email'];
    storepassword = data['store_password'];
    return data;
  }
}
